domain/cloud/service/service.go

```diff
@@ -21,17 +30,26 @@ type State interface {
 
 	// ListClouds returns the clouds matching the optional filter.
 	ListClouds(context.Context, string) ([]cloud.Cloud, error)
+
+	// WatchCloud returns a new NotifyWatcher watching for changes to the specified cloud.
+	WatchCloud(
```

Comment:
Watching looks to be a concern of the service layer not the state layer. Just based on all the other watchers. I think here is would make sense to have the state layer just expose `CloudUUIDForName` functionality, then the watching is only a concern of the service layer.

Reply:
We only have cloud name at the service layer. We watch db changes keyed on uuid, and uuid is a state layer concern. We need to do it this way if we want to avoid surfacing db layer concerns outside the state layer.

Reply:
I can follow up and confirm we like this pattern or not, and change if necessary. It's all below the service later anyway.